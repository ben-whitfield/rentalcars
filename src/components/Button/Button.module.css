.button {
  position: absolute;
  right: 0;
  width: auto;
  font: inherit;
  padding: 0.5rem 1.5rem;
  border: 1px solid #068323;
  color: white;
  background: #068323;
  box-shadow: 0 0 4px rgba(0, 0, 0, 0.26);
  cursor: pointer;
  display: inline-block;
  font-weight: 700;
  height: 64px;
  border-top-right-radius: 6px;
  border-bottom-right-radius: 6px;
}

.button:focus {
  outline: none;
}

.button:hover,
.button:active {
  background: #0d6521;
  border-color: #0d6521;
  box-shadow: 0 0 8px rgba(0, 0, 0, 0.26);
}