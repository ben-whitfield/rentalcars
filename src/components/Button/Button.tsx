import React from 'react';
import styles from './Button.module.css';

interface ButtonProps {
    /** Which type of button */
    type: 'submit'|'button'|'reset'
    /** The text of the button */
    text: string
}

const Button = ({ type, text }: ButtonProps) => {
    return <button className={styles.button} type={type}>{text}</button>;
};

export default Button;
