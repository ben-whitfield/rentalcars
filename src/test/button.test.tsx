import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../components/Button/Button';

describe('Button', () => {
  test('snapshot render button', () => {
    const component = renderer.create(<Button type='submit' text='Search'  />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});