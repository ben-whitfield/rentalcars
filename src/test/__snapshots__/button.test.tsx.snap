// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot render button 1`] = `
<button
  className="button"
  data-testid="searchButton"
  type="submit"
>
  Search
</button>
`;
